File: backend/app/schemas/branch.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime
from typing import Optional
from decimal import Decimal

class BranchBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    code: str = Field(..., min_length=1, max_length=50)
    is_hq: bool = False

    address: Optional[str] = None
    city: Optional[str] = None
    province: Optional[str] = None
    postal_code: Optional[str] = None
    country: str = "Indonesia"
    phone: Optional[str] = None
    email: Optional[str] = None

    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None

    timezone: str = "Asia/Jakarta"
    currency: str = "IDR"

class BranchCreate(BranchBase):
    pass

class BranchUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    code: Optional[str] = Field(None, min_length=1, max_length=50)
    is_hq: Optional[bool] = None

    address: Optional[str] = None
    city: Optional[str] = None
    province: Optional[str] = None
    postal_code: Optional[str] = None
    country: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None

    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None

    timezone: Optional[str] = None
    currency: Optional[str] = None
    is_active: Optional[bool] = None

class BranchInDB(BranchBase):
    id: UUID
    tenant_id: UUID
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True

class BranchResponse(BranchInDB):
    pass

class BranchListResponse(BaseModel):
    branches: list[BranchResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/branch.ts
export interface Branch {
  id: string;
  tenant_id: string;
  name: string;
  code: string;
  is_hq: boolean;

  address?: string;
  city?: string;
  province?: string;
  postal_code?: string;
  country: string;
  phone?: string;
  email?: string;

  latitude?: number;
  longitude?: number;

  timezone: string;
  currency: string;

  is_active: boolean;
  created_at: string;
  updated_at?: string;
}

export interface BranchCreate {
  name: string;
  code: string;
  is_hq?: boolean;
  address?: string;
  city?: string;
  province?: string;
  postal_code?: string;
  country?: string;
  phone?: string;
  email?: string;
  latitude?: number;
  longitude?: number;
  timezone?: string;
  currency?: string;
}

export interface BranchUpdate {
  name?: string;
  code?: string;
  is_hq?: boolean;
  address?: string;
  city?: string;
  province?: string;
  postal_code?: string;
  country?: string;
  phone?: string;
  email?: string;
  latitude?: number;
  longitude?: number;
  timezone?: string;
  currency?: string;
  is_active?: boolean;
}

export interface BranchListResponse {
  branches: Branch[];
  total: number;
  page: number;
  page_size: number;
}
